src/ships.js

```diff
@@ -0,0 +1,36 @@
+class Ship {
+    constructor(itinerary) {
+
+        this.itinerary = itinerary;
+        this.currentPort = itinerary.ports[0];
+        this.previousPort = null;
+        this.currentPort.addShip(this);
+    };
+
+    setSail() {
+        this.currentPort.removeShip(this);
+
+        const itinerary = this.itinerary;
+
+        const currentPortIndex = itinerary.ports.indexOf(this.currentPort);
+
+        if (currentPortIndex === (itinerary.ports.length - 1)) {
+            throw new Error(`End of itinerary reached`);
+        };
+
+        this.previousPort = this.currentPort;
+        this.currentPort = null;
+    };
+
+    dock() {
+
+        const itinerary = this.itinerary;
```

Comment:
Same as above. Since it already exists in the constructor and it's not too verbose, use it directly.